brain/CMakeLists.txt

```diff
@@ -8,16 +8,22 @@ source_group(\\ FILES CMakeLists.txt)
 
 set(BRAIN_PUBLIC_HEADERS
   circuit.h
-  morphology.h
   types.h
+  cell/morphology.h
+  cell/section.h
+  cell/soma.h
```

Comment:
As far as I know, they are different enough to make it difficult to provide base classes for both. Besides that I don't know if there's any use case that requires the base class.